wp-content/themes/core/components/blocks/gallery_grid/Gallery_Grid_Controller.php

```diff
@@ -0,0 +1,346 @@
+
+		$main_attrs['data-swiper-options'] = $swiper_options;
+
+		$slider = [
+			Slider_Controller::SLIDES          => $this->get_slides(),
+			Slider_Controller::SHOW_CAROUSEL   => false,
+			Slider_Controller::SHOW_ARROWS     => true,
+			Slider_Controller::SHOW_PAGINATION => false,
+			Slider_Controller::MAIN_ATTRS      => $main_attrs,
+			Slider_Controller::CLASSES         => [ 'b-gallery-grid__slider' ],
+		];
+
+		return $slider;
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_gallery_img_ids(): array {
+		$gallery_ids = [];
+
+		if ( empty( $this->gallery ) ) {
+			return $gallery_ids;
+		}
+
+		foreach ( $this->gallery as $img ) {
+			if ( empty( $img ) ) {
+				return [];
+			}
+			$gallery_ids[] = $img[ 'id' ];
+		}
+
+		return $gallery_ids;
+	}
+
+	/**
+	 * @param int $slide_id
+	 *
+	 * @return string|Deferred_Component
+	 */
+	protected function get_image_caption( int $slide_id ) {
+		$thumbnail_image = get_posts( [ 'p' => $slide_id, 'post_type' => 'attachment' ] );
+
+		if ( empty( $thumbnail_image ) && empty( $thumbnail_image[0] ) ) {
+			return '';
+		}
+
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::CLASSES => [
+				'b-gallery__slide-caption',
+			],
+			Text_Controller::CONTENT => esc_html( $thumbnail_image[0]->post_excerpt ) ?? '',
+		] );
+	}
+
+	/**
+	 * @param $img_id
+	 *
+	 * @return string
+	 */
+	protected function get_image_template( $index, $img_id ): string {
+		$img          = wp_get_attachment_image( $img_id, Image_Sizes::CORE_FULL );
+		$slide_markup = $img;
+		$slide_markup .= '
+			<div class="b-gallery-grid__meta-wrap">
+				<div class="b-gallery-grid__meta-count">' . ( $index + 1 ) . __(' OF ', 'tribe') . count( $this->get_gallery_img_ids() ) . '</div>
+				<div class="b-gallery-grid__meta-caption">' . $this->get_image_caption( $img_id ) . '</div>
+			</div>';
+
+		return $slide_markup;
+	}
+
+	/**
+	 * Get Slides
+	 *
+	 * @return array
+	 */
+	protected function get_slides(): array {
+		$slide_ids = $this->get_gallery_img_ids();
+		$slides    = [];
+
+		if ( empty( $slide_ids ) ) {
+			return $slides;
+		}
+
+		return array_map( function ( $index, $slide_id ) {
+			return $this->get_image_template( $index, $slide_id );
+		}, array_keys($slide_ids), $slide_ids );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_gallery_img_thumbs(): array {
+		$ids          = $this->get_gallery_img_ids();
+		$gallery_imgs = [];
+		$i            = 1;
+		$img_size     = Image_Sizes::SQUARE_MEDIUM;
+		$img_bg       = true;
+		$img_bg_class = 'c-image--bg';
+		$img_aspect   = 's-aspect-ratio-1-1';
+
+		if ( empty( $ids ) ) {
+			return $gallery_imgs;
+		}
+
+		if ( $this->grid_layout === self::ONE ) {
+			$img_size     = Image_Sizes::CORE_FULL;
```

Comment:
I worry that this, too, will be too large for mobile devices. I think we should use both a `src_size` size and `srcset_sizes` so we don't unintentionally load over-sizes images.